core/kazoo_services/src/kz_services_tasks.erl

```diff
@@ -0,0 +1,152 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2016, 2600Hz
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%   Pierre Fenoll
+%%%-------------------------------------------------------------------
+-module(kz_services_tasks).
+%% behaviour: tasks_provider
+
+-export([help/0
+        ,category/0
+        ,module/0
+        ,output_header/1
+        ]).
+
+%% Verifiers
+-export([
+        ]).
+
+%% Appliers
+-export([descendant_quantities/2
+        ]).
+
+-include("kazoo_services.hrl").
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec category() -> ne_binary().
+category() -> <<"services">>.
+
+-spec module() -> module().
+module() -> kz_util:to_binary(?MODULE).
+
+-spec output_header(atom()) -> kz_csv:row().
+output_header('descendant_quantities') ->
+    [<<"account_id">>
+    ,<<"month">>
+    ,<<"category">>
+    ,<<"item">>
+    ,<<"quantity_bom">>
+    ,<<"quantity_eom">>
+    ].
+
+-spec help() -> kz_proplist().
+help() ->
+    [{<<"descendant_quantities">>
+     ,kz_json:from_list([{<<"description">>, <<"List per-month descendant accounts quantities">>}
+                        ])
+     }
+    ].
+
+%%% Verifiers
+
+
+%%% Appliers
+
+-spec descendant_quantities(kz_proplist(), task_iterator()) -> task_iterator().
+descendant_quantities(Props, 'init') ->
+    Descendants = get_descendants(props:get_value('auth_account_id', Props)),
+    DescendantsMoDBs = lists:flatmap(fun kapps_util:get_account_mods/1, Descendants),
+    lager:debug("found ~p descendants & ~p MoDBs in total"
+               ,[length(Descendants), length(DescendantsMoDBs)]),
+    {'ok', DescendantsMoDBs};
+
+descendant_quantities(_, []) ->
+    'stop';
+
+descendant_quantities(_, [SubAccountMoDB | DescendantsMoDBs]) ->
+    ?MATCH_MODB_SUFFIX_ENCODED(A, B, Rest, Year, Month) = SubAccountMoDB,
+    AccountId = ?MATCH_ACCOUNT_RAW(A, B, Rest),
+    YYYYMM = <<Year/binary, Month/binary>>,
+    BoM = modb_service_quantities(SubAccountMoDB, 'true'),
+    EoM = modb_service_quantities(SubAccountMoDB, 'false'),
+    Data =
```

Comment:
I'd also like to see the 'work' portion broken out into its own function and a unit test made to show what is being built here. This will aid in refactoring, I think.